libsql-wal/src/shared_wal.rs

```diff
@@ -20,7 +20,7 @@ use libsql_sys::name::NamespaceName;
 
 #[derive(Default)]
 pub struct WalLock {
-    pub(crate) tx_id: Arc<Mutex<Option<u64>>>,
+    pub(crate) tx_id: Arc<async_lock::Mutex<Option<u64>>>,
```

Comment:
why async lock over the tokio one? Is it for the blocking stuff?

Reply:
eventually, I would like to make Warin runtime agnostic, so I try to avoid tokio primitives outside of the IO trait